WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift

```diff
@@ -692,11 +692,16 @@ extension OrderListViewController: UITableViewDelegate {
         let order = orderDetailsViewModel.order
         ServiceLocator.analytics.track(event: WooAnalyticsEvent.Orders.orderOpen(order: order))
         selectedOrderID = order.orderID
+        let allViewModels = allViewModels()
+        let currentIndex = allViewModels.firstIndex(where: { $0.order.orderID == order.orderID })
+
+        guard let currentIndex = currentIndex else { return }
 
         if isSplitViewInOrdersTabEnabled {
-            switchDetailsHandler(orderDetailsViewModel)
+            splitViewController?.isCollapsed ?? true ? switchDetailsHandler(allViewModels, currentIndex) :
+            switchDetailsHandler([orderDetailsViewModel], 0)
```

Comment:
Same as above, the logic here is hard to follow 😅 , could we make it more readable with some switch or re-declaring these locally?

Reply:
Yeah, keeping both the split view logic and single view is tricky, I made it more readable with comments in 8af2c0033d2441508d7c90da9b082a2508f0f84d